Code/Mantid/Framework/Algorithms/test/ChangeTimeZeroTest.h

```diff
@@ -0,0 +1,571 @@
+    do_act_and_assert(relativeTimeShift, absoluteTimeShift,
+                      inputWorkspaceName, outputWorkspaceName);
+
+    // Clean up
+    cleanUpWorkspaces(inputWorkspaceName, outputWorkspaceName);
+  }
+
+  void test_changed_time_for_standard_setting_and_relative_fractional_time_and_same_inOutWS() {
+    // Arrange
+    const std::string inputWorkspaceName = "inWS";
+    const std::string outputWorkspaceName = inputWorkspaceName;
+    const std::string absoluteTimeShift = "";
+    const double relativeTimeShift = 1020.5;
+    Mantid::API::MatrixWorkspace_sptr ws =
+        provideEventWorkspace(LogType::STANDARD, inputWorkspaceName);
+
+    // Act and assert
+    do_act_and_assert(relativeTimeShift, absoluteTimeShift,
+                      inputWorkspaceName, outputWorkspaceName);
+
+    // Clean up
+    cleanUpWorkspaces(inputWorkspaceName, outputWorkspaceName);
+  }
+
+
+  // WorkspaceSingleValue tests
+  void test_changed_time_for_standard_setting_and_relative_time_and_differnt_inOutWS_and_WorkspaceSingleValue() {
+    // Arrange
+    const std::string inputWorkspaceName = "inWS";
+    const std::string outputWorkspaceName = "outWS";
+    const std::string absoluteTimeShift = "";
+    const double relativeTimeShift = 1000;
+    Mantid::API::MatrixWorkspace_sptr ws =
+      provideWorkspaceSingleValue(LogType::STANDARD, inputWorkspaceName);
+
+    // Act and assert
+    do_act_and_assert(relativeTimeShift, absoluteTimeShift,
+                      inputWorkspaceName, outputWorkspaceName);
+
+    // Clean up
+    cleanUpWorkspaces(inputWorkspaceName, outputWorkspaceName);
+  }
+
+  void test_changed_time_for_standard_setting_and_absolute_time_and_differnt_inOutWS_and_WorkspaceSingleValue() {
+    // Arrange
+    const std::string inputWorkspaceName = "inWS";
+    const std::string outputWorkspaceName = "outWS";
+    DateAndTime absoluteTimeShift(m_startTime);
+    absoluteTimeShift += 1000.0;
+    const double relativeTimeShift = 0.0;
+    Mantid::API::MatrixWorkspace_sptr ws =
+      provideWorkspaceSingleValue(LogType::STANDARD, inputWorkspaceName);
+
+    // Act and assert
+    do_act_and_assert(relativeTimeShift, absoluteTimeShift.toISO8601String(),
+                      inputWorkspaceName, outputWorkspaceName);
+
+    // Clean up
+    cleanUpWorkspaces(inputWorkspaceName, outputWorkspaceName);
+  }
+
+private:
+  DateAndTime m_startTime;
+  DateAndTime m_stringPropertyTime;
+  boost::shared_ptr<Mantid::Kernel::DateTimeValidator> m_dateTimeValidator;
+  int m_length;
+  std::string m_doubleSeriesID;
+  std::string m_boolSeriesID;
+  std::string m_intSeriesID;
+  std::string m_stringSeriesID;
+  std::string m_stringID;
+  std::string m_comparisonWorkspaceName;
+
+  enum LogType { STANDARD, NOPROTONCHARGE };
+
+  // act and assert
+  void do_act_and_assert(double relativeTimeShift, std::string absoluteTimeShift,
+                         std::string inWsName, std::string outWsName) {
+      // Act
+      ChangeTimeZero alg;
```

Comment:
If possible, always run the algorithm as a child, that way you have proper scope rules and don't have to fiddle around with the ADS.